main.cpp

```diff
@@ -1,14 +1,66 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include "draw.h"
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <cstdlib>
+#include "include/config.h"
+#include "include/abstractrule.h"
+#include "include/boardgraph.h"
+#include "include/textrenderer.h"
 
-int main()
+using namespace std;
+
+const int CLI_ARG_ERROR_CODE = 1;
+const int CONFIG_PARSE_ERROR_CODE = 2; 
+
+// Command line arguments
+// -c [Config file path]
+// -r [text|image]
+int main(int argv, char **argc)
 {
-    if (Catan::Draw::Initialize())
-    {
+  srand(time(NULL));
+	Catan::Generate::Config config;
+	string renderType = "text";
+	// Handle command line arguments
```

Comment:
if you didn't know, you can handle command line arguments in C and C++ w/ getopt